app/src/main/java/com/sun_asterisk/moviedb_50/screen/details/MovieDetailsPresenter.kt

```diff
@@ -37,7 +39,77 @@ class MovieDetailsPresenter(private val movieRepository: MovieRepository) :
                         onGetMovieTrailerSuccess(data.trailers)
                     }
                     view?.onLoading(true)
+                    isFavoriteMovie(movieID.toString())
                 }
             })
     }
+
+    private fun isFavoriteMovie(movieID: String) {
+        movieRepository.findFavoriteId(movieID, object : OnDataLoadedCallback<Boolean> {
+            override fun onSuccess(data: Boolean?) {
+                data ?: return
+                view?.showFavoriteImage(
+                    if (data) R.drawable.ic_favorite else R.drawable.ic_favorite_border
+                )
+            }
+
+            override fun onError(e: Exception) {
+                view?.onError(e)
+            }
+
+        })
+    }
+
+    override fun handleFavorites(favorite: Favorite) {
+        movieRepository.findFavoriteId(favorite.movieID, object : OnDataLoadedCallback<Boolean> {
+            override fun onSuccess(data: Boolean?) {
+                data ?: return
+                if (data) deleteFavorite(favorite.movieID) else addFavorite(favorite)
+            }
+
+            override fun onError(e: Exception) {
+                view?.onError(e)
+            }
+        })
+    }
+
+    private fun addFavorite(favorite: Favorite) {
+        movieRepository.addFavorite(favorite, object : OnDataLoadedCallback<Boolean> {
+            override fun onSuccess(data: Boolean?) {
+                data ?: return
+                view?.run {
+                    if (data) {
```

Comment:
nếu data null thì sao 
```suggestion
                    if (data == true) {
```